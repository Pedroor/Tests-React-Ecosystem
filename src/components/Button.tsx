import React from 'react';

// import { Container } from './styles';

interface Props {
    children: React.ReactNode;
    onClick: () => void;
}

const Button: React.FC<Props> = ({ children, onClick }) => {
  return <button onClick={onClick}><b>{children}</b></button>;
}

export default Button;