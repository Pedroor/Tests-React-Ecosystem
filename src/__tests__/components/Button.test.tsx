import React from 'react'
import { render, screen } from '@testing-library/react'
import Button from '../../components/Button'
import userEvent from '@testing-library/user-event'

describe('Test button successfully', () =>{
    it('Render Button', () =>{
        const buttonText = 'Fazer login'
       const { container }= render(<Button>{buttonText}</Button>)

       const button = screen.getByRole('button', { name: buttonText})
       expect(button).toBeInTheDocument();
       expect(container).toMatchSnapshot();
    })

    it('sucess onClick function called', () =>{
        const buttonText = 'Fazer login'
        const onClick = jest.fn();

       const { container }= render(<Button onClick={onClick}>{buttonText}</Button>)

       const button = screen.getByRole('button', { name: buttonText})

       userEvent.click(button);
       
       expect(onClick).toHaveBeenCalledTimes(1);
    
    })
})

// Testes de regress√£o visual