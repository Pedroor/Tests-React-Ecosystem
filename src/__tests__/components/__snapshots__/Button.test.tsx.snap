// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button successfully Render Button 1`] = `
<div>
  <button>
    <b>
      Fazer login
    </b>
  </button>
</div>
`;
